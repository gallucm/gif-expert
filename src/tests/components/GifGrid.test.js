const { shallow } = require("enzyme");
const { GifGrid } = require("../../components/GifGrid");
const { useFetchGifs } = require("../../hooks/useFetchGifs");
jest.mock('../../hooks/useFetchGifs'); // mock de llamada

describe('Pruebas para <GifGrid/>', () => {
    
    const category = 'Goku';

    const gifs = [{
        id: 'ABC',
        url: 'http://www.pruebascr.com/hola.jpeg',
        title: 'cualquieru titulo'
    }]

    useFetchGifs.mockReturnValue({
        data: gifs,
        loading: true
    });

    const wrapper = shallow(<GifGrid category={category} />);    
   
    test('Debe mostrar items cuando se cargan las imagenes con useFetchGifs', () => {
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    });
    

})
