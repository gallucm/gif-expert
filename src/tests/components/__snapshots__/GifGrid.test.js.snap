// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas para <GifGrid/> Debe mostrar items cuando se cargan las imagenes con useFetchGifs 1`] = `
<Fragment>
  <br />
  <br />
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="cualquieru titulo"
      url="http://www.pruebascr.com/hola.jpeg"
    />
  </div>
</Fragment>
`;
